scripts/inline_comments_data_collection.py

```diff
@@ -0,0 +1,172 @@
+
+
+class NoDiffsFoundException(Exception):
+    def __init__(self, patch_id):
+        super().__init__(f"No diffs found for the given patch ID: {patch_id}")
+        self.patch_id = patch_id
+
+
+class NoTransactionsFoundException(Exception):
+    def __init__(self, patch_id):
+        super().__init__(f"No transactions found for the given patch ID: {patch_id}")
+        self.patch_id = patch_id
+
+
+class NoDiffFoundForPHIDException(Exception):
+    def __init__(self, phid):
+        super().__init__(f"No diff found for PHID {phid}")
+        self.phid = phid
+
+
+def find_revision_from_patch(patch_id):
+    diffs = api.search_diffs(diff_id=patch_id)
+
+    if not diffs:
+        raise NoDiffsFoundException(patch_id)
+
+    revision_phid = diffs[0]["revisionPHID"]
+    return revision_phid
+
+
+def find_transactions_from_patch(patch_id):
+    revision_phid = find_revision_from_patch(patch_id)
+    transactions = api.request("transaction.search", objectIdentifier=revision_phid)[
+        "data"
+    ]
+
+    if not transactions:
+        raise NoTransactionsFoundException(patch_id)
+
+    return transactions
+
+
+def get_diff_info_from_phid(phid):
+    diffs = api.search_diffs(diff_phid=phid)
+    if not diffs:
+        raise NoDiffFoundForPHIDException(phid)
+    return diffs[0]["id"], diffs[0]["revisionPHID"]
+
+
+def find_bugid_from_revision_phid(phid):
+    revision = api.load_revision(rev_phid=phid)
+    return revision["fields"]["bugzilla.bug-id"]
+
+
+def download_inline_comments():
+    for patch_id, comments in review_data.get_all_inline_comments(lambda c: True):
+        save_comments_to_file(patch_id, comments)
+    return
+
+
+def save_comments_to_file(patch_id, comments):
+    resolved_comments = [comment for comment in comments if comment.is_done]
+
+    file_path = f"comments/{patch_id}.json"
+    if os.path.exists(file_path) or not resolved_comments:
+        return
+
+    with open(file_path, "w") as f:
+        json.dump([comment.__dict__ for comment in resolved_comments], f, indent=4)
+
+
+def find_recent_update(transactions, comment_date_modified):
+    updates = [
+        transaction
+        for transaction in transactions
+        if transaction["type"] == "update"
+        and transaction["dateModified"] <= comment_date_modified
+    ]
+    if not updates:
+        return None
+    most_recent_update = max(
+        updates, key=lambda transaction: transaction["dateModified"]
+    )
+    return most_recent_update
+
+
+def save_to_dataset(data):
+    dataset_file = "dataset/inline_comment_dataset.json"
+    with open(dataset_file, "a") as f:
+        f.write(json.dumps(data) + "\n")
+
+
+def to_int(value):
+    if not value:
+        return None
+    if not isinstance(value, int):
+        return int(value)
+    return value
+
+
+def process_comments(patch_threshold, diff_length_threshold):
+    comments_dir = "comments"
+    patch_count = 0
+    for file_name in os.listdir(comments_dir):
```

Comment:
Why not directly rely on `review_data.get_all_inline_comments()` instead of reading from disk?

Reply:
Similar to the other comments, I've removed any disk reading here: 3fffb4900f54ffa5799a7b465e73d027b6496d6e